src/Model/Message/MessageBag.php

```diff
@@ -4,19 +4,26 @@
 
 namespace PhpLlm\LlmChain\Model\Message;
 
-/**
- * @template-extends \ArrayObject<int, MessageInterface>
- */
-final class MessageBag extends \ArrayObject implements \JsonSerializable
+final class MessageBag implements \Countable, \JsonSerializable
 {
+    /**
+     * @var MessageInterface[]
+     */
+    public array $messages = [];
+
     public function __construct(MessageInterface ...$messages)
     {
-        parent::__construct(array_values($messages));
+        $this->messages = array_values($messages);
+    }
+
+    public function add(MessageInterface $message): void
```

Comment:
what about
```suggestion
    public function add(MessageInterface ...$messages): void
```
to support adding multiple messages?

Reply:
Hmm the use case is quite unusual I would say.